Sources/PubNub/Helpers/Crypto/CryptoModule.swift

```diff
@@ -273,6 +275,48 @@ public struct CryptoModule {
     return streamDecryptionResult
   }
 
+  /// Decrypts the stream from the given URL and writes it to the output path
+  ///
+  /// - Parameters:
+  ///   - url: URL of the encrypted stream
+  ///   - outputPath: Path to write the decrypted stream
+  /// - Returns:
+  ///  - **Success**: A decrypted `InputStream` object
+  ///  - **Failure**: `PubNubError` describing the reason of failure
+  @discardableResult
+  public func decryptStream(from url: URL, to outputPath: URL) -> Result<InputStream, PubNubError> {
+    PubNub.log.debug(
+      "Decrypting file",
+      category: .crypto
+    )
+
+    guard let inputStream = InputStream(url: url) else {
+      PubNub.log.debug(
+        "Cannot create InputStream from \(url). Ensure that the file exists at the specified path",
+        category: .crypto
+      )
+      return .failure(PubNubError(
+        .decryptionFailure,
+        additional: ["File doesn't exists at \(url) path"]
+      ))
+    }
+
+    let streamDecryptionResult = performStreamDecryption(
+      stream: inputStream,
+      contentLength: url.sizeOf,
+      to: outputPath
+    )
+
```

Comment:
Shouldn't we add:
```
 inputStream.open()
  defer { inputStream.close() }
```

The thing is that you’re creating the InputStream from disk, but you never open it (or close it on all paths). A simple inputStream.open()/defer { inputStream.close() } around your decrypt call will ensure the file handle is managed correctly.

Reply:
This is the URLSession object that will open and close the stream when uploading it to the remote. It's not the CryptoModule that will manage it